File: py/cli/src/tuish_cli/commands/logout.py
from __future__ import annotations

import typer
from rich.console import Console

from tuish_cli.config import delete_config
from tuish_cli.output import emit_success

console = Console()


def logout(ctx: typer.Context) -> None:
    delete_config(ctx.obj.config_path)
    if ctx.obj.json:
        emit_success("Logged out successfully")
        return
    console.print("[green]Credentials cleared.[/green]")
    console.print(f"[dim]Config: {ctx.obj.config_path}[/dim]")

File: rs/cli/src/commands/logout.rs
use super::{delete_config, output_json, CliResult, Context};
use colored::Colorize;
use serde_json;

pub fn run(ctx: &Context) -> CliResult {
	delete_config(&ctx.config_path)?;
	if ctx.json {
		let payload = serde_json::json!({
			"success": true,
			"message": "Logged out successfully",
		});
		return output_json(&payload);
	}
	println!("{}", "Credentials cleared.".green());
	println!("{}", format!("Config: {}", ctx.config_path.display()).dimmed());
	Ok(())
}
